acm/validator/bucket.go

```diff
@@ -66,15 +66,17 @@ func (vc *Bucket) AlterPower(id crypto.PublicKey, power *big.Int) (*big.Int, err
 			"in a total power greater than that allowed by tendermint (%v): would make next total power: %v",
 			errHeader, id.GetAddress(), vc.Previous.GetPower(id.GetAddress()), power, maxTotalVotingPower, nextTotalPower)
 	}
+
 	// If we call vc.flow.ChangePower(id, absFlow) (below) will we induce a change in flow greater than the allowable
 	// flow we have left to spend?
-	if vc.Flow.Flow(id, absFlow).Cmp(allowableFlow) == 1 {
+	if vc.Flow.Flow(id, absFlow).Cmp(allowableFlow) == 1 && allowableFlow.Cmp(big.NewInt(0)) > 0 {
```

Comment:
hm why is this needed? I think if allowableFlow is negative we have a bug, no?

Reply:
`maxFlow` is `-1` if there are no validators

Reply:
As a bare minimum this needs a comment. But I don't particularly like handling it this way anyway. It's a special case so we should treat it as a special case rather than relying on this trick of the light.

In fact this is an important edge case of max flow...

Reply:
We may want something like this, but I'm not sure yet, need to sleep on it: https://github.com/gregdhill/burrow/pull/2